core/src/main/java/org/infinispan/configuration/parsing/Parser.java

```diff
@@ -105,15 +105,19 @@
    public Parser() {
    }
 
+   private void addJGroupsDefaultStacksIfNeeded(final XMLExtendedStreamReader reader, final ConfigurationBuilderHolder holder) {
+      if (holder.getJGroupsStack(BuiltinJGroupsChannelConfigurator.TCP_STACK_NAME) == null) {
```

Comment:
It isn't directly related. I will create a JIRA for it separate it out. This was found because adding a new cache starts up the parser and reads these darn stack files every time. I wasn't sure how to actually fix it, and this is a nice easy way I thought (should make startup time without clustering even faster as well).

Reply:
You mean starting a new `DefaultCacheManager`, right? If we start the parser every time we add a cache in an existing manager, that's another bug that needs to be fixed.

Reply:
Nope. But I might be mixing words, it might have been just that it ran the parser, not started. Either way it parses these jgroups files every time.

Reply:
I don't know what the difference between running and starting the parser is, I assumed you were talking about `readElement()`.

I put a breakpoint in `readElement()` and we definitely don't hit that when we use the programmatic configuration to create a cache, but we hit it when we create a cache via the admin API, because the configuration that gets replicated is a full `<infinispan/>` XML. Maybe that's what you were seeing?

Reply:
Yes, it was via admin API. Sorry I did this so long ago I forgot all the details.

Reply:
That's why it's good to create the JIRA before the PR :P